src/pspm_load_data.m

```diff
@@ -327,163 +298,22 @@
   filestruct.posofmarker = filestruct.posofmarker(1); % first marker channel
 end
 %% 9 Return channels, or save file
-if isstruct(channel)
-  infos = channel.infos;
-  data = channel.data;
-end
-flag = zeros(numel(data), 1);
-if ischar(channel) && ~strcmp(channel, 'none')
-  if contains(channel,'pupil')
-    if strcmpi(channel, 'pupil') && isfield(infos.source, 'best_eye')
-      flag = get_chans_to_load_for_pupil(data, infos.source.best_eye, 0);
-    elseif strcmpi(channel(7), 'l') || strcmpi(channel(7), 'r')
-      flag = get_chans_to_load_for_pupil(data, channel(7), 1);
-    end
-  elseif strcmpi(channel, 'sps') && isfield(infos.source, 'best_eye')
-    flag = get_chans_to_load_for_sps(data, infos.source.best_eye);
-  else
-    for k = 1:numel(data)
-      if (any(strcmpi(channel, {'event', 'events'})) && ...
-          strcmpi(data{k}.header.units, 'events')) || ...
-          (strcmpi(channel, 'wave') && ~strcmpi(data{k}.header.units, 'events')) || ...
-          (any(strcmpi(channel, {'trigger', 'marker'})) && ...
-          any(strcmpi(data{k}.header.chantype, {'trigger', 'marker'})))
-        flag(k) = 1;
-      elseif strcmp(data{k}.header.chantype, channel)
-        flag(k) = 1;
-      end
+if ischar(channel) && strcmp(channel, 'none')
```

Comment:
The code is simplified because only one channel will be processed instead of all channels.. is that correct?

Reply:
There still can be multiple channels. But I moved the eyetracker channel loading logic to pspm_load_channel, so it is removed here.